pages/css-grid-superposition--qcj5jo/css-grid-superposition--qcj5jo.md

```diff
@@ -0,0 +1,279 @@
+---
+title: Prevent layout shift with CSS grid superposition
+date: Last Modified
+layout: article.njk
+---
+
+<!-- TODO still some confusion between "we" and I -->
+
+There are two reasons why people use CSS grid:
+
+1. 😎 CSS is awesome! It's a fact, deal with it. 
+1. 🛠️ Grid is a great tool to build complex two-dimensional layouts.
+
+I sometimes have a third reason to use CSS grid, I call it the "anti layout shift superposition technique".
+Let me explain with some real examples.
+
+**NOTE:** This article is focused on layout shift occuring when toggling state on components.
+We won't cover layout shift happening on first page load.
+
+## Let's get some context
+
+I work for [Clever Cloud](https://www.clever-cloud.com/en/), an IT automation platform.
+Our clients push their code and we handle the rest for them: build, deploy, host, scale, maintain, recover...
+When they need to configure their applications, databases and other services, they use our web UI: "the console".
+I spend most of my time on this project and it looks like this:
+
+![Console overview of a Node.js application](./console-overview.png)
+
+What you see in the screenshot above is the overview of my own site, a Node.js application hosted on our platform.
+On the right, you can see a pie chart which represents the repartition of HTTP response codes returned by the application in the last 24 hours.
+
+![UI Component displaying a chart](./cc-tile-status-codes-chart.png)
+
+<!-- TODO -->
+When I worked on this component, I imagined our users would implicitly understand the chart is related to the current application.
+I decided to move some details like the exact number of requests in tooltips accessible via mouse and touch.
+The problem was, how do I explain the following details:
+
+* The chart only displays data from the last 24 hours.
+* The legend items can be clicked to show/hide the different status code categories. Thank you [Chart.js](https://www.chartjs.org/)!
+
+The available space was too small to add a title at the top or bottom of the chart so to help our users figure this out, I added an information button ℹ️ in the top right corner.
```

Comment:
Isn't there a title already?

Also, Make that two sentences: "The available space… of the chart. To help our users figure…"
(Above ~18 words, a sentence is bloated.)

Reply:
There is a title but it does not specify the range of the data.

Maybe this works better (with the split sentence as well):

```suggestion
The available space was too small to have a detailed title at the top or bottom of the chart. To help our users figure this out, I added an information button ℹ️ in the top right corner.
```

Reply:
I guess I meant a chart.js title, in the canvas etc, like we have on requests at the bottom.
I'll fix that, it's confusing.